ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/tabsontop/v1/tabsontop/clientlibs/site/js/tabs.js

```diff
@@ -269,34 +266,50 @@
-                            tabs[i].setAttribute(FormView.Constants.TABINDEX, "0");
-                        } else {
-                            tabpanels[i].classList.remove(Tabs.selectors.active.tabpanel);
-                            tabpanels[i].setAttribute(FormView.Constants.ARIA_HIDDEN, true);
-                            tabs[i].classList.remove(Tabs.selectors.active.tab);
-                            tabs[i].setAttribute(FormView.Constants.ARIA_SELECTED, false);
-                            tabs[i].setAttribute(FormView.Constants.TABINDEX, "-1");
-                        }
+                for (var i = 0; i < tabpanels.length; i++) {
+                    if (tabs[i].id === this._active) {
+                        tabpanels[i].classList.add(Tabs.selectors.active.tabpanel);
+                        tabpanels[i].removeAttribute(FormView.Constants.ARIA_HIDDEN);
+                        tabs[i].classList.add(Tabs.selectors.active.tab);
+                        tabs[i].setAttribute(FormView.Constants.ARIA_SELECTED, true);
+                        tabs[i].setAttribute(FormView.Constants.TABINDEX, "0");
+                    } else {
+                        tabpanels[i].classList.remove(Tabs.selectors.active.tabpanel);
+                        tabpanels[i].setAttribute(FormView.Constants.ARIA_HIDDEN, true);
+                        tabs[i].classList.remove(Tabs.selectors.active.tab);
+                        tabs[i].setAttribute(FormView.Constants.ARIA_SELECTED, false);
+                        tabs[i].setAttribute(FormView.Constants.TABINDEX, "-1");
                     }
-                } else {
-                    // only one tab
-                    tabpanels.classList.add(Tabs.selectors.active.tabpanel);
-                    tabs.classList.add(Tabs.selectors.active.tab);
                 }
             }
         }
 
+        #getBeforeViewElement(instanceManager,instanceIndex) {
+            var result={};
+            var instanceManagerId=instanceManager.getId();
+            if(instanceIndex==0){
+                var closestNonRepeatableFieldId=this._templateHTML[instanceManagerId]['closestNonRepeatableFieldId'];
+                var closestRepeatableFieldInstanceManagerId=this._templateHTML[instanceManagerId]['closestRepeatableFieldInstanceManagerId'];
+                var indexToInsert=this.#getTabIndexToInsert(closestNonRepeatableFieldId,closestRepeatableFieldInstanceManagerId);
+                var tabPanels=this.#getCachedTabPanels();
+                if(indexToInsert>0){
+                    result.beforeViewElement = this.#getTabPanelElementById(tabPanels[indexToInsert - 1].id);
+                }else{
+                    result.beforeViewElement = this.#getTabListElement();
+                }
+            }else {
+                var previousInstanceElement=instanceManager.children[instanceIndex-1].element;
+                result.beforeViewElement = previousInstanceElement.parentElement.parentElement;
```

Comment:
can you pls put a comment in code on why it's grand parent and not immediate parent.